### PR TITLE
[14.0][ADD] auth_saml_enviroment

### PR DESCRIPTION
**Pull request by leemannd**
Base work and functionality is present.

Improvement to be added:
- update the readme

Depends on https://github.com/OCA/server-auth/pull/268

### CODE REVIEW

File: server_environment/models/server_env_mixin.py [hunk 1]
```diff
@@ -190,6 +190,7 @@ def _server_env_read_from_config(self, field_name, config_getter):
             # at this point we should have checked that we have a key with
             # _server_env_has_key_defined so we are sure that the value is
             # either in the global or the record config
+            print(field_name, config_getter)
```
Comment (leemannd) [inline server_environment/models/server_env_mixin.py]: Todo remove it

**Review by yvaucher — APPROVED** [review]
LGTM

just a print to remove and doc to fill

File: server_environment/models/server_env_mixin.py (hunk 1 above)
Comment (yvaucher) [inline server_environment/models/server_env_mixin.py]: :eyes:

File: auth_saml_environment/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def _list_saml_providers_domain(self):
+        """OVERWRITE domain to return all active IDP
+
+        The configuration of an IDP doesn't rely anymore on
+        sp_pem_public and sp_pem_private as those could be
+        set through sp_pem_private_path and sp_pem_public_path.
+
+        Keeping it simple by relying only on the active field.
+        """
+        return []
```
Comment (theangryangel) [inline auth_saml_environment/controllers/main.py]: Upstream auth_saml basically does the same thing now, so I think this isn't needed any more? Beyond that it works great in our test environment :)

**Review by theangryangel — CHANGES_REQUESTED** [review]

**Comment by leemannd** [comment]
The linked module has been merged. Could we consider merging it? cc @yvaucher

**Comment by simahawk** [comment]
@leemannd build is red. Also, pls squash commit when done

**Review by SilvioC2C — APPROVED** [review]
LGTM

File: auth_saml_environment/tests/test_environment_variable.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            "sp_pem_public_path=file1.txt\n"
+            "sp_pem_private_path=file2.txt"
+        )
+        with self.set_config_dir(None), self.set_env_variables(env_var):
+            parser = server_env._load_config()
+            self.assertEqual(
+                list(parser.keys()), ["DEFAULT", "auth_saml_provider.sample"]
+            )
+            self.assertDictEqual(
+                dict(parser["sample"].items()),
```
Comment (yvaucher) [inline auth_saml_environment/tests/test_environment_variable.py]: Unit test gives a `KeyError: 'sample'` on this line.

Shouldn't it be `auth_saml_provider.sample` ?

Comment (leemannd) [reply auth_saml_environment/tests/test_environment_variable.py]: Fixed. Thanks

**Comment by leemannd** [comment]
@simahawk @yvaucher Tests fixed. Can be merged

**Pull request merged**